src/DataTypes/Serializations/SerializationArray.cpp

```diff
@@ -514,6 +514,33 @@ static ReturnType deserializeTextImpl(IColumn & column, ReadBuffer & istr, Reade
     return ReturnType(true);
 }
 
+void SerializationArray::readArraySafe(DB::IColumn & column, std::function<void()> && read_func)
+{
+    size_t initial_size = column.size();
+
+    try
+    {
+        read_func();
```

Comment:
Can't we pass columns to `read_func` instead of capturing them in outer scope?

Reply:
We can, I don't have a strong opinion what is better, if you think it will be better to pass columns as arguments, I will do it